### ISSUE TITLE
Release 2016.07 - RC1

### ISSUE THREAD

**Issue by kYc0o** [RIOT-OS/Release-Specs#21]
This issue lists the status of all tests for the [Release Candidate 1 of the 2016.07](https://github.com/RIOT-OS/RIOT/commit/b5cb68bd6673abe02166f8af7782aa2745ca08cd) release.

Specs tested:
- [x] [01-CI](https://github.com/RIOT-OS/Release-Specs/blob/master/01-ci/01-ci.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/01-ci/01-ci.md#task-01) [SUCCESS]
 - [x] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/01-ci/01-ci.md#task-02) [SUCCESS]
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/01-ci/01-ci.md#task-03) [SUCCESS]
 - [x] [Task 04](https://github.com/RIOT-OS/Release-Specs/blob/master/01-ci/01-ci.md#task-04) [SUCCESS]
- [x] [02-Tests](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md#task-01) [SUCCESS, cf. https://github.com/RIOT-OS/Release-Specs/issues/22]
 - [x] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md#task-02) [SUCCESS, cf. https://github.com/RIOT-OS/Release-Specs/issues/23]
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md#task-03) [SUCCESS, cf. https://github.com/RIOT-OS/Release-Specs/issues/24]
- [x] [03-Single Hop IPv6 ICMP](https://github.com/RIOT-OS/Release-Specs/blob/master/03-single-hop-ipv6-icmp/03-single-hop-ipv6-icmp.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/03-single-hop-ipv6-icmp/03-single-hop-ipv6-icmp.md#task-01)
 - [x] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/03-single-hop-ipv6-icmp/03-single-hop-ipv6-icmp.md#task-02)
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/03-single-hop-ipv6-icmp/03-single-hop-ipv6-icmp.md#task-03)
- [ ] [04-Single Hop 6LoWPAN ICMP](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md#task-01) [SUCCESS]
 - [ ] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md#task-02) [FAIL https://github.com/RIOT-OS/RIOT/issues/5486]
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md#task-03) [SUCCESS]
 - [x] [Task 04](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md#task-04) [SUCCESS]
- [ ] [05-Single Hop Route](https://github.com/RIOT-OS/Release-Specs/blob/master/05-single-hop-route/05-single-hop-route.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/05-single-hop-route/05-single-hop-route.md#task-01) [SUCCESS]
 - [ ] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/05-single-hop-route/05-single-hop-route.md#task-02) [FAIL https://github.com/RIOT-OS/RIOT/issues/5486]
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/05-single-hop-route/05-single-hop-route.md#task-03) [SUCCESS]
- [x] [06-Single Hop UDP](https://github.com/RIOT-OS/Release-Specs/blob/master/06-single-hop-udp/06-single-hop-udp.md) [SUCCESS]
- [ ] [07-Multi Hop](https://github.com/RIOT-OS/Release-Specs/blob/master/07-multi-hop/07-multi-hop.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/07-multi-hop/07-multi-hop.md#task-01) [SUCCESS]
 - [x] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/07-multi-hop/07-multi-hop.md#task-02) [SUCCESS]
 - [x] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/07-multi-hop/07-multi-hop.md#task-03) [SUCCESS]
 - [x] [Task 04](https://github.com/RIOT-OS/Release-Specs/blob/master/07-multi-hop/07-multi-hop.md#task-04) [SUCCESS]
- [ ] [08-Interop](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md)
 - [x] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-01)
 - [x] [Task 02](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-02)
 - [ ] [Task 03](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-03)
 - [x] [Task 04](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-04)
 - [x] [Task 05](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-05)
 - [x] [Task 06](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-06)
 - [x] [Task 07](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-07)
 - [ ] [Task 08](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-08) [FAIL https://github.com/RIOT-OS/RIOT/pull/5718 + interop problems from lwIP's side]
 - [x] [Task 09](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-09)
 - [ ] [Task 10](https://github.com/RIOT-OS/Release-Specs/blob/master/08-interop/08-interop.md#task-10) [FAIL https://github.com/RIOT-OS/RIOT/pull/5718 + interop problems from lwIP's side]

**Comment by BytesGalore** [comment]
Currently performing 
- [ ] [02-Tests](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests/02-tests.md)
 - [ ] [Task 01](https://github.com/RIOT-OS/Release-Specs/blob/master/02-tests)

Observations:
- `driver_dht` compiles without warning runs and obviously fails
- `driver_pir` compiles without warning runs and obviously fails on GPIO access
- `emb6` automatically sets samr21-xpro as target and fails on missing compiler 

more to come :)

**Comment by BytesGalore** [comment]
I made #22 to discuss collect observations for `02-Tests/Task 01`

**Comment by kYc0o** [comment]
Great! Thanks! :D

**Comment by PeterKietzmann** [comment]
I will start with with 03-Single Hop IPv6 ICMP

**Comment by PeterKietzmann** [comment]
I will continue with 04-Single Hop 6LoWPAN ICMP

**Comment by kYc0o** [comment]
Can you edit the comment to update if the tests were successful?

**Comment by PeterKietzmann** [comment]
04-Single Hop 6LoWPAN ICMP Task 03 gets stuck

**Comment by PeterKietzmann** [comment]
(tested with iotlab-m3 to iotlab-m3 and iotlab-m3 to samr21-xpro)

**Comment by miri64** [comment]
Uhm, where is the tag to test this against?

**Comment by kYc0o** [comment]
I didn't do it, can you? I don't know if I have rights.

**Comment by PeterKietzmann** [comment]
Just for the record: I tested with commit `b5cb68bd6673abe02166f8af7782aa2745ca08cd`

**Comment by miri64** [comment]
Mh come to think of it @OlegHahm you want to wait for https://github.com/RIOT-OS/RIOT/pull/2071 to be merged before we create 2016.07-branch (on which RC1 would be)... so let's wait for it.

**Comment by kYc0o** [comment]
Yes @PeterKietzmann  that's the last commit before the last PR was merged this afternoon. @miri64  I'd like to wait for that one too.

**Comment by miri64** [comment]
@kYc0o and when we backport it?

**Comment by kYc0o** [comment]
Performing Task 01

**Comment by PeterKietzmann** [comment]
Performing 06-Single Hop UDP now on the pre-RC

**Comment by BytesGalore** [comment]
I made #23 and #24 for the remaining 02-Tests/*

**Comment by kYc0o** [comment]
CI task 02 works only with launchpad versions of arm-none-eabi-gcc, and fails with embedded versions at least on Debian Jessie and Ubuntu 14.04.

**Comment by BytesGalore** [comment]
02-Tests succeeded

**Comment by kYc0o** [comment]
I'm going through 07

**Comment by PeterKietzmann** [comment]
04/Task 03 and 05/Task 02 are failing on a node (iotlab-m3). It seems the great payload size (1KB) is the problem

**Comment by miri64** [comment]
@PeterKietzmann at which side did you do your experiments. I had high packet loss in Grenoble because the nodes picked by default by the cli-tools are very close to a running experiment. Just tried 04/Task 03 in Grenoble and hand no problems (still running but should be >90% received)

**Comment by miri64** [comment]
Now it's done: it was it 0% packet loss in Lyon.

**Comment by PeterKietzmann** [comment]
Hmm. I've tested in Paris as well as on my desk.

**Comment by kYc0o** [comment]
Sorry I'm also using grenoble for RPL experiments. However theres someone abusing on the usage, maybe he's generating traffic and interferences.

**Comment by PeterKietzmann** [comment]
And you tried with `ping6 100 <addy> 1000 0 10`?

**Comment by PeterKietzmann** [comment]
PS: Paris was nearly unused at that time

**Comment by miri64** [comment]
No `And you tried with ping6 100 <addy> 1024 10` is the correct one. Forgot, what stats interval is, but its not needed for these tests.

**Comment by miri64** [comment]
> Sorry I'm also using grenoble for RPL experiments. However theres someone abusing on the usage, maybe he's generating traffic and interferences.

@kYc0o for small payloads background traffic is no problem, but for 1024 B (when your IPv6 packet consists of >10 radio frames) or even smaller (in my tests problems started at around 550 B) the packet loss is naturally much higher.

**Comment by miri64** [comment]
Just went into it with the debugger and found that for both 04/Task 03 and 05/Task 02 https://github.com/RIOT-OS/RIOT/issues/5486 is at least one of the reasons for the problems @PeterKietzmann (and now me too) are seeing. For the release I would recommend to note it down as a known problem.

**Comment by miri64** [comment]
(unless someone feels like fixing)

**Comment by miri64** [comment]
One can check with the debugger btw by typing

```
(gdb) print at86rf2xx_devs[0].pending_tx
```

If the result isn't zero while nothing is sending the state is broken and interrupts got lost.

**Comment by kYc0o** [comment]
Ok let's put it in the Release notes as a known issue, hoping to solve it for the next release.

**Comment by PeterKietzmann** [comment]
BTW: would you share a link to the WIP release notes? Maybe link it somewhere in the issue description above?

**Comment by kYc0o** [comment]
There is [one](http://www.yourpart.eu/p/RIOT-RN2016.07) since the beginning that I'd be sooo glad to get help with :)

**Comment by miri64** [comment]
I did the RIOT<=>RaspPi tests on a samr21-xpro, since my IoT-Lab nodes don't seem to be usable anymore (terminal does not react :(). Should be valid anyway.

**Comment by miri64** [comment]
Since I'm the one who introduced emb6 and lwIP I think its best if someone else does 8.8-10. @BytesGalore, what about you? You used the applications in Task 2 already ;-).

**Comment by miri64** [comment]
Oh and @kYc0o can you please do 8.3?

**Comment by kYc0o** [comment]
Yes I'll do it asap

**Comment by miri64** [comment]
I think tomorrow would suffice ;-)

**Comment by kYc0o** [comment]
I'd like to wait for #5698 before making lwip tests but it seems to fail with Murdock...

**Comment by PeterKietzmann** [comment]
@kYc0o #5698 is not a bugfix. Why should it go into the release branch?

**Comment by kYc0o** [comment]
Yes I know it's not a bugfix but lwip is intended to run on native, which would be able to compile the tests and if it fails we should fix it... But we can continue like this if you think it's not really am obstacle.

**Comment by PeterKietzmann** [comment]
I had some problems testing 08/08 and 08/10. For GNRC I used the default examples/gnrc_networking and for lwip tests/lwip. I sniffed the traffic in parallel to see what is going on.

1.1 GNRC to lwip
GNRC sends UDP packets but lwip does not react.

1.2 lwip to GNRC
Instead of directly sending the UDP packet to the link local address, the lwip sends a neighbour solicitation (NS) message to the GNRC node. GNRC answers with four neighbour advertisements (NA) (probably L2 retransmissions?). That NS, NA sequence occurs three times.

2.1 emb6 to lwip
emb6 simply sends out a UDP packet but the lwip does not (visibly) react on that.

2.2 lwip to emb6
Instead of simply sending a UDP packet, the lwip node sends a NA on which the em6 node responds with a NS. That happens three times 

Additionally I tried lwip to lwip. That worked as expected. There one can see a sequence of NS, NA, UDP.

@miri64 "promised" to have a look at it later.

**Comment by PeterKietzmann** [comment]
@kYc0o if #5698 is needed to perform task 01/02 it would be fine with me.

**Comment by kYc0o** [comment]
> I did the RIOT<=>RaspPi tests on a samr21-xpro, since my IoT-Lab nodes don't seem to be usable anymore (terminal does not react :(). Should be valid anyway.

Yes of course, the goal is to test interoperability between RIOT and RaspPi, so with a samr21 it's also valid.

**Comment by miri64** [comment]
So lwIP is the only one that is problematic? Do I see that right?

**Comment by PeterKietzmann** [comment]
From what I reported it seems that interoperability with lwip is the critical part, yes.

**Comment by miri64** [comment]
Okay, I digged into it now and basically this is the problem:

* this is the nodes EUI-64 `5a:5a:5c:5c:87:92:76:0e`
* and this is the generated link-local address from it: `fe80::5c5c:5a5a:e76:9287`

So somehow the bytes of the address get quite mixed up :-/, so the backtranslation RIOT does for link-local addresses does not work. I'll try to find out if that's a problem with my glue-code or a problem with lwIP itself. For now I would using vanilla-NDP with 6LoWPAN for interop with lwIP (which is "easy" with GNRC if you know your way around the dependencies `gnrc_ipv6_router_default` pulls in). No idea, why emb6 has a problem with that, but I suspect they implement this part of RFC 6775.

**Comment by miri64** [comment]
Okay another problem is that while parsing the IEEE 802.15.4 header lwIP is [just "skipping" the source PAN](http://git.savannah.gnu.org/cgit/lwip.git/tree/src/netif/lowpan6.c?id=92f385aaed7dcccde632502711281234ccda958c#n1005) not checking if source PAN compression is activated. So this is a bug on lwIP's side. For the order of the EUI-64 bytes I'll open a PR regardless, but I would just create a known issue, that lwIP's 6LoWPAN support is pretty insular for now.

**Comment by miri64** [comment]
though... that's a known issue for lwIP... not for us I would say.

**Comment by miri64** [comment]
For PR https://github.com/RIOT-OS/RIOT/pull/5718.

**Comment by kYc0o** [comment]
Can we reference the failing tests to their corresponding issues?

**Comment by miri64** [comment]
Done

**Comment by kYc0o** [comment]
Great! so I need to finish the RPL stuff and then try Contiki interop and we should be done! :D

**Comment by PeterKietzmann** [comment]
Any progress?

**Comment by kYc0o** [comment]
No but I think I'll give up... I'll finish the release notes today and make the release announcement.

**Comment by miri64** [comment]
I'd like to have a look at the release notes, too and maybe find time to help in the afternoon.

**Comment by kYc0o** [comment]
OK let's wait to the afternoon to make the release ;)
